lib/galaxy/schema/invocation.py

```diff
@@ -214,3 +231,114 @@ class InvocationMessageResponseModel(BaseModel):
 
     class Config:
         orm_mode = True
+
+
+class InvocationState(str, Enum):
+    NEW = "new"  # Brand new workflow invocation... maybe this should be same as READY
+    READY = "ready"  # Workflow ready for another iteration of scheduling.
+    SCHEDULED = "scheduled"  # Workflow has been scheduled.
+    CANCELLED = "cancelled"
+    FAILED = "failed"
+
+
+class InvocationStepState(str, Enum):
+    NEW = "new"  # Brand new workflow invocation step
+    READY = "ready"  # Workflow invocation step ready for another iteration of scheduling.
+    SCHEDULED = "scheduled"  # Workflow invocation step has been scheduled.
+    # CANCELLED = 'cancelled',  TODO: implement and expose
+    # FAILED = 'failed',  TODO: implement and expose
+
+
+class InvocationStepOutput(Model):
+    src: str = Field(
+        literal_to_value(INVOCATION_STEP_OUTPUT_SRC),
+        title="src",
+        description="The source model of the output.",
+        const=True,
+        mark_required_in_schema=True,
+    )
```

Comment:
The type of `src` should be `INVOCATION_STEP_OUTPUT_SRC` instead of `str` that way it would be considered as a constant, so you probably don't need `const=true` and `mark_required_in_schema=True`.